### PR TITLE
Refactored CommandHandler

### PR DESCRIPTION
**Pull request by volundmush**
#### Brief overview of PR changes/additions
Rewrote the CommandHandler into a replaceable Class, and get_and_merge_cmdsets to be naive of the kinds of objects (session, account, object) it is working with.

Shifted logic for gathering "local object" cmdsets from the get_and_merge_cmdset function to the cmd-bearing typeclasses as obj.get_extra_cmdsets()

#### Motivation for adding to Evennia
Noticed that CmdHandler isn't very flexible or easy to extend and that the cmdset merge process could be made far cleaner, simpler, and more flexible.

Would like to add more kinds of objects that can bear Commands in user-code and this helps a ton with that.

#### Other info (issues closed, discussion etc)
I poked at the tests but I'm not sure that they're all truly fixed.

### CODE REVIEW

**Comment by volundmush** [comment]
After more futzing I can tell this particular approach is not going to work.

Still curious about your thoughts on the overall notion.

File: evennia/commands/cmdhandler.py [hunk 1]
```diff
@@ -377,63 +320,29 @@ def _get_cmdsets(obj):
                 _msg_err(caller, _ERROR_CMDSETS)
                 raise ErrorReported(raw_string)
             try:
-                returnValue((obj.cmdset.current, list(obj.cmdset.cmdset_stack)))
+                if extra:
+                    extra = yield obj.get_extra_cmdsets(caller, current, cmdsets)
+                    returnValue(extra)
```
Comment (Griatch) [inline evennia/commands/cmdhandler.py:325]: I think in latest Twisted, `returnValue` is not actually needed anymore, and a regular `return value` now works from inside an `@inlineCallbacks`-decorated function.. But I've not actually tested it ...

**Review by Griatch — COMMENTED** [review]
I don't mind making the CmdHandler more pluggable by all means, but as you notice, it's not so easy to just change it around. Having individual entities be responsible for finding cmdsets on objects is an interesting approach too. But overall, it's probably a good idea to make sure unit tests pass before making the PR ;)

**Comment by Griatch** [comment]
With no more comments or activity after that last comment, I'm closing this.

**Pull request closed without merging**